torch/_dynamo/variables/builder.py

```diff
@@ -899,6 +899,21 @@ def wrap_tensor(self, value: torch.Tensor):
         # then the relevant SubgraphTracer will lift it to being an input of
         # the subgraph.
         # See NOTE [HigherOrderOperator tracing design] for more details.
+
+        if not self.tx.output.export:
+            # Export has (supposedly) valid cases for fake tensors as inputs here.
+            # I am not convinced, atm, but out of scope for what this assert was added for (protecting value checks
+            # in real_value_tensor_positive_aliases in the common case)
```

Comment:
The PR in question is https://github.com/pytorch/pytorch/pull/100017 I recommend reading through it. I will say that wanting to do a PT2 export without actually instantiating the model with real parameters seems like a legitimate use case to me.

Reply:
Thats fair, but for that case I suppose we don't care about these guards as much, and can probably just not have the dedup and aliasing guards for export...